Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 6e7629bd6c4353a36b9a4719752eaadff2b4b57c
Number of Error(s): 5
/mnt/disk1/home/vamosax/smoketest/PR-12435/HPCC-Platform/rtl/eclrtl/rtlfield.cpp:907:10: error: prototype for \xe2\x80\x98unsigned int RtlKeyedIntTypeInfo::queryChildType() const\xe2\x80\x99 does not match any in class \xe2\x80\x98RtlKeyedIntTypeInfo\xe2\x80\x99
/mnt/disk1/home/vamosax/smoketest/PR-12435/HPCC-Platform/rtl/eclrtl/rtlfield.hpp:219:33: error: candidate is: virtual const RtlTypeInfo* RtlKeyedIntTypeInfo::queryChildType() const
make[2]: *** [rtl/eclrtl/CMakeFiles/eclrtl.dir/rtlfield.cpp.o] Error 1
make[1]: *** [rtl/eclrtl/CMakeFiles/eclrtl.dir/all] Error 2
make: *** [all] Error 2

[Build error:](https://gist.github.com/e482b97e29b0c697ed2492cf8ba07270)
Build log [Build log](https://gist.github.com/HPCCSmoketest/e482b97e29b0c697ed2492cf8ba07270#file-result-19-04-09-14-56-08-log)

--- a/rtl/eclrtl/rtlfield.cpp
+++ b/rtl/eclrtl/rtlfield.cpp
@@ -904,6 +904,12 @@ __int64 RtlKeyedIntTypeInfo::removeBias(unsigned __int64 value, unsigned length)
     return value - ((unsigned __int64)1 << (length*8-1));
 }
 
+const RtlTypeInfo * RtlKeyedIntTypeInfo::queryChildType() const
+{
+    return child;
+}
+
+
 //-------------------------------------------------------------------------------------------------------------------
 
 size32_t RtlPackedIntTypeInfo::getMinSize() const

--- a/rtl/eclrtl/rtlfield.hpp
+++ b/rtl/eclrtl/rtlfield.hpp
@@ -216,6 +216,8 @@ struct ECLRTL_API RtlKeyedIntTypeInfo final : public RtlTypeInfoBase
     virtual int compare(const byte * left, const byte * right) const override;
     virtual bool canMemCmp() const override { return true; }
     virtual unsigned hash(const byte *self, unsigned inhash) const override;
+    virtual const RtlTypeInfo * queryChildType() const override;
+
 private:
     inline __uint64 getUInt(const void * ptr) const { return (__uint64) getInt(ptr); }
     static unsigned __int64 addBias(__int64 value, unsigned length);

